h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstTfIdf.java

```diff
@@ -0,0 +1,153 @@
+package water.rapids.ast.prims.advmath;
+
+import hex.tfidf.DocumentFrequency;
+import hex.tfidf.InverseDocumentFrequency;
+import hex.tfidf.TermFrequency;
+import hex.tfidf.TfIdfPreprocessor;
+import water.Key;
+import water.MRTask;
+import water.fvec.*;
+import water.rapids.Env;
+import water.rapids.Merge;
+import water.rapids.Val;
+import water.rapids.ast.AstPrimitive;
+import water.rapids.ast.AstRoot;
+import water.rapids.vals.ValFrame;
+
+/**
+ * Primitive AST operation to compute TF-IDF values for given document corpus.
+ */
+public class AstTfIdf extends AstPrimitive<AstTfIdf> {
+
+    /**
+     * Name to be used for a column containing Inverse Document Frequency values in the output frame of this operation.
+     */
+    private static final String IDF_COL_NAME = "IDF";
+    /**
+     * Name to be used for a column containing TF-IDF values in the output frame of this operation.
+     */
+    private static final String TF_IDF_COL_NAME = "TF-IDF";
+    /**
+     * Column names to be used for preprocessed frame.
+     */
+    private static final String[] PREPROCESSED_FRAME_COL_NAMES = new String[] { "DocID", "Word" };
+
+    @Override
+    public int nargs() {
+        return 1 + 2; // (tf-idf input_frame_name preprocess)
+    }
+
+    @Override
+    public String[] args() {
+        return new String[]{ "frame", "preprocess" };
+    }
+
+    @Override
+    public Val apply(Env env, Env.StackHelp stk, AstRoot[] asts) {
+        Frame inputFrame = stk.track(asts[1].exec(env).getFrame());
```

Comment:
What if a user will put a frame with more than two columns?

Reply:
I added check for the input types - `Num`, `Str` and number of columns (`>= 2`). Or would you prefer to have parameters to indicate which columns should be used?

I also added corresponding unit tests for this, and added check for empty input.

Reply:
I think, the parameter for indicating the columns should be the best way. WDYT @honzasterba  and @michalkurka ?

Reply:
sounds good to me, something like this:

`tf_idf(frame, document_id=(str, int), document_word=(str,int), preprocess, case_sens)`

I think expecting that the frame has some pre-defined structure or column names is not something we do very often in H2O

Reply:
I changed it to expect column indices for `AstTfIdf` and for `R` and `Python` methods both column names and indices are allowed.